lib_xcore_math/src/arch/ref/vect_headroom.c

```diff
@@ -14,6 +14,10 @@ headroom_t vect_s16_headroom(
     const int16_t v[],
     const unsigned length)
 {
+    // If an array of zero size is given,
+    // return the maximum headroom
+    if (length == 0) return 15;
+
```

Comment:
As far as I can tell, assembly does not do the same here. I'm not sure why we need to support this case.

Reply:
This problem exists only in Windows. We check an initialized array and it should always be zero in XS3a and Linux, this is not the case on Windows.
In my opinion it is a bug to use uninititalized memory and we should fix it in assembly too.

Reply:
In https://github.com/xmos/lib_xcore_math/pull/156/commits/41d6a2160864428d692ff60e306aeb2604858f22 I removed the checks and I updated all the vect tests so that the minimum length is 0.